pkg/pillar/cmd/downloader/resolveconfig.go

```diff
@@ -0,0 +1,194 @@
+package downloader
+
+import (
+	"errors"
+	"fmt"
+	"net"
+
+	zconfig "github.com/lf-edge/eve/api/go/config"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/zedUpload"
+	log "github.com/sirupsen/logrus"
+)
+
+// Handles both create and modify events
+func handleAppImgResolveModify(ctxArg interface{}, key string,
+	configArg interface{}) {
+
+	ctx := ctxArg.(*downloaderContext)
+	config := configArg.(types.ResolveConfig)
+	log.Infof("handleAppImgResolveModify for %s\n", key)
+	resolveTagsToHash(ctx, config)
+	log.Infof("handleAppImgResolveModify for %s, done\n", key)
+}
+
+func handleAppImgResolveDelete(ctxArg interface{}, key string,
+	configArg interface{}) {
+	ctx := ctxArg.(*downloaderContext)
+	config := configArg.(types.ResolveConfig)
+	ctx.pubAppImgResolveStatus.Unpublish(config.Key())
+	log.Infof("handleAppImgResolveDelete for %s\n", key)
+}
+
+func publishResolveStatus(ctx *downloaderContext,
+	status *types.ResolveStatus) {
+
+	key := status.Key()
+	log.Debugf("publishResolveStatus(%s)\n", key)
+	pub := ctx.pubAppImgResolveStatus
+	pub.Publish(key, *status)
+	log.Debugf("publishResolveStatus(%s) Done\n", key)
+}
+
+func unpublishResolveStatus(ctx *downloaderContext,
+	status *types.ResolveStatus) {
+
+	key := status.Key()
+	log.Debugf("unpublishResolveStatus(%s)\n", key)
+	pub := ctx.pubAppImgResolveStatus
+	pub.Unpublish(key)
+	log.Debugf("unpublishResolveStatus(%s) Done\n", key)
+}
+
+func lookupResolveConfig(ctx *downloaderContext,
+	key string) *types.ResolveConfig {
+
+	sub := ctx.subAppImgResolveConfig
+	c, _ := sub.Get(key)
+	if c == nil {
+		log.Infof("lookupResolveConfig(%s) not found\n", key)
+		return nil
+	}
+	config := c.(types.ResolveConfig)
+	return &config
+}
+
+func lookupResolveStatus(ctx *downloaderContext,
+	key string) *types.ResolveStatus {
+
+	pub := ctx.pubAppImgResolveStatus
+	c, _ := pub.Get(key)
+	if c == nil {
+		log.Infof("lookupResolveStatus(%s) not found\n", key)
+		return nil
+	}
+	status := c.(types.ResolveStatus)
+	return &status
+}
+
+func resolveTagsToHash(ctx *downloaderContext, rc types.ResolveConfig) {
+	var (
+		err                           error
+		errStr, remoteName, serverURL string
+		syncOp                        zedUpload.SyncOpType = zedUpload.SyncOpGetObjectMetaData
+		trType                        zedUpload.SyncTransportType
+		auth                          *zedUpload.AuthInput
+	)
+
+	rs := lookupResolveStatus(ctx, rc.Key())
+	if rs == nil {
+		rs = &types.ResolveStatus{
+			AppID:        rc.AppID,
+			ImageID:      rc.ImageID,
+			Name:         rc.Name,
+			PurgeCounter: rc.PurgeCounter,
+		}
+	}
+
+	dst, errStr := lookupDatastoreConfig(ctx, rc.DatastoreID, rc.Name)
```

Comment:
Would make sense to insert the code here to see if there is already an @sha in the name, and if so extract that sha and return it.

Reply:
Yeah, it makes sense. If the name has SHA then we are extracting and returning from here.